adam-cli/src/main/scala/org/bdgenomics/adam/cli/CompareADAM.scala

```diff
@@ -173,13 +173,19 @@ class CompareADAM(protected val args: CompareADAMArgs) extends ADAMSparkCommand[
 
   def run(sc: SparkContext, job: Job): Unit = {
 
-    if (args.listComparisons) {
+    // Work around for -list_comparisons (doesn't require @Arguments)
+    if (!args.listComparisons && (args.input1Path == null || args.input2Path == null)) {
+      val options = new CompareADAMArgs
+      val parser = new CmdLineParser(options)
+      println("\n2 Arguments (files) must be given as input.\n")
+      parser.printUsage(System.err)
+      sys.exit(1)
```

Comment:
You might want to compare this to how Args4j handles parsing errors. It might raise an exception instead of calling System.exit(). If that is the case, that behavior should be emulated.